src/components/reusable/daterangepicker/daterangepicker.ts

```diff
@@ -0,0 +1,288 @@
+
+  /** Date warning text */
+  @property({ type: String })
+  warnText = '';
+
+  /** Maximum date in YYYY-MM-DD format.
+   * If the value isn't a possible date string in the format, then the element has no maximum date value
+   */
+  @property({ type: String })
+  maxDate = '';
+
+  /** Minimum date in YYYY-MM-DD format,
+   * If the value isn't a possible date string in the format, then the element has no minimum date value.
+   */
+  @property({ type: String })
+  minDate = '';
+
+  /** Specifies the granularity that the value must adhere to, or the special value any,
+   * For date inputs, the value of step is given in days; and is treated as a number of milliseconds equal to 86,400,000 times the step value.
+   * The default value of step is 1, indicating 1 day.*/
+  @property({ type: String })
+  step = '';
+
+  override render() {
+    return html`
+      <label
+        class="datepicker-label-text"
+        for=${this.name}
+        ?disabled=${this.disabled}
+      >
+        ${this.required ? html`<span class="required">*</span>` : null}
+        <slot></slot>
+      </label>
+
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-1` : 'date-range-1'}
+          name=${this.name}
+          value=${this.startDate}
+          ?required=${this.required}
+          ?disabled=${this.disabled}
+          ?invalid=${this.invalidText !== ''}
+          min=${ifDefined(this.minDate)}
+          max=${ifDefined(this.maxDate)}
+          step=${ifDefined(this.step)}
+          @input=${(e: any) => this.handleStartDate(e)}
+        />
+      </div>
+
+      <span class="range-span">—</span>
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-2` : 'date-range-2'}
+          value=${this.endDate}
+          ?disabled=${this.disabled}
+          ?invalid=${this.invalidText !== ''}
+          min=${ifDefined(this.startDate ?? '')}
+          max=${ifDefined(this.maxDate)}
+          step=${ifDefined(this.step)}
+          @input=${(e: any) => this.handleEndDate(e)}
+        />
+      </div>
+
+      ${this.caption !== ''
+        ? html` <div class="caption">${this.caption}</div> `
+        : null}
+      ${this.invalidText !== ''
+        ? html` <div class="error">${this.invalidText}</div> `
+        : null}
+      ${this.warnText !== '' && this.invalidText === ''
+        ? html`<div class="warn">${this.warnText}</div>`
+        : null}
+    `;
+  }
+
+  override updated(changedProps: PropertyValues) {
+    if (changedProps.has('startDate')) {
+      // set form data value
+      this.internals.setFormValue('start-date', this.startDate);
```

Comment:
Use `this.name + '-start'` here instead of 'start-date'.

Reply:
I have doubt here. again what happens if user not entered **name** `@property` ? if **name** property is required then how we can force user to use this **name** prop ?

Reply:
Yeah I don't think we can really enforce it to be required, just mention that it is. The reason is, start-date and end-date need to be unique, since there could be more than one on a page/form. Hence why "name" is required for all form inputs.

Reply:
ok